ui/common/MiningStatisticsHash.tsx

```diff
@@ -0,0 +1,418 @@
+              hashPower > 27000,
+          })}
+        ></div>
+        <div
+          className={clsx('h-[15px] w-[5px] rounded-full bg-black/25', {
+            '!bg-[#FBEBC6] shadow-[0_0px_8px_rgba(251,235,198,0.5)]':
+              hashPower > 28000,
+          })}
+        ></div>
+        <div
+          className={clsx('h-[15px] w-[5px] rounded-full bg-black/25', {
+            '!bg-[#FBEBC6] shadow-[0_0px_8px_rgba(251,235,198,0.5)]':
+              hashPower > 29000,
+          })}
+        ></div>
+        <div
+          className={clsx('h-[15px] w-[5px] rounded-full bg-black/25', {
+            '!bg-[#FBEBC6] shadow-[0_0px_8px_rgba(251,235,198,0.5)]':
+              hashPower > 30000,
+          })}
+        ></div>
+        <div
+          className={clsx('h-[15px] w-[5px] rounded-full bg-black/25', {
+            '!bg-[#FBEBC6] shadow-[0_0px_8px_rgba(251,235,198,0.5)]':
+              hashPower > 31000,
+          })}
+        ></div>
+        <div
+          className={clsx('h-[15px] w-[5px] rounded-full bg-black/25', {
+            '!bg-[#FBEBC6] shadow-[0_0px_8px_rgba(251,235,198,0.5)]':
+              hashPower > 32000,
+          })}
+        ></div>
+        <div
+          className={clsx('h-[15px] w-[5px] rounded-full bg-black/25', {
+            '!bg-[#FBEBC6] shadow-[0_0px_8px_rgba(251,235,198,0.5)]':
+              hashPower > 33000,
+          })}
+        ></div>
+        <div
+          className={clsx('h-[15px] w-[5px] rounded-full bg-black/25', {
+            '!bg-[#FBEBC6] shadow-[0_0px_8px_rgba(251,235,198,0.5)]':
+              hashPower > 34000,
+          })}
+        ></div>
+        <div
+          className={clsx('h-[15px] w-[5px] rounded-full bg-black/25', {
+            '!bg-[#FBEBC6] shadow-[0_0px_8px_rgba(251,235,198,0.5)]':
+              hashPower > 35000,
+          })}
+        ></div>
+        <div
+          className={clsx('h-[15px] w-[5px] rounded-full bg-black/25', {
+            '!bg-[#FBEBC6] shadow-[0_0px_8px_rgba(251,235,198,0.5)]':
+              hashPower > 36000,
+          })}
+        ></div>
+        <div
+          className={clsx('h-[15px] w-[5px] rounded-full bg-black/25', {
+            '!bg-[#FBEBC6] shadow-[0_0px_8px_rgba(251,235,198,0.5)]':
+              hashPower > 37000,
+          })}
+        ></div>
+        <div
+          className={clsx('h-[15px] w-[5px] rounded-full bg-black/25', {
+            '!bg-[#FBEBC6] shadow-[0_0px_8px_rgba(251,235,198,0.5)]':
+              hashPower > 38000,
+          })}
+        ></div>
+        <div
+          className={clsx('h-[15px] w-[5px] rounded-full bg-black/25', {
+            '!bg-[#FBEBC6] shadow-[0_0px_8px_rgba(251,235,198,0.5)]':
+              hashPower > 39000,
+          })}
+        ></div>
+        <div
+          className={clsx('h-[15px] w-[5px] rounded-full bg-black/25', {
+            '!bg-[#FBEBC6] shadow-[0_0px_8px_rgba(251,235,198,0.5)]':
+              hashPower > 40000,
+          })}
+        ></div>
```

Comment:
I was analyzing the code and noticed this section of repetitive code. 

Using `.map()` will be a good idea here.

You can define a const array of numbers at the beginning of the code.

```js
const hashDiff = Array.from({length: 40}, (_, i) => i + 1)
```

And use it to map these divs.

```suggestion
    {hashDiff.map((i, index) => {
          return (
            <div key={index}
              className={clsx('h-[15px] w-[5px] rounded-full bg-black/25', {
                '!bg-[#FBEBC6] shadow-[0_0px_8px_rgba(251,235,198,0.5)]':
                  hashPower > i * 1000,
              })}
            ></div>
          )
        })}
```

Reply:
Thanks for the suggestion. Using a map will be more optimized and will require less work if in the future we want to change the specification of the loading bar/ progress bar. I will try to convert it into a reusable component